src/py21cmfast/io/caching.py

```diff
@@ -232,13 +233,13 @@ class RunCache:
     """
 
     InitialConditions: Path = _pathfield()
-    PerturbedField: tuple[Path] = _dict_of_paths_field()
-    TsBox: tuple[Path] = _dict_of_paths_field()
-    IonizedBox: tuple[Path] = _dict_of_paths_field()
-    BrightnessTemp: tuple[Path] = _dict_of_paths_field()
-    HaloBox: tuple[Path] = _dict_of_paths_field()
-    PerturbHaloField: tuple[Path] = _dict_of_paths_field()
-    XraySourceBox: tuple[Path] = _dict_of_paths_field()
+    PerturbedField: dict[float, Path] = _dict_of_paths_field()
+    TsBox: dict[float, Path] = _dict_of_paths_field()
+    IonizedBox: dict[float, Path] = _dict_of_paths_field()
+    BrightnessTemp: dict[float, Path] = _dict_of_paths_field()
+    HaloBox: dict[float, Path] | None = _dict_of_paths_field()
+    PerturbHaloField: dict[float, Path] | None = _dict_of_paths_field()
+    XraySourceBox: dict[float, Path] | None = _dict_of_paths_field()
```

Comment:
I'm guessing since this is constructed directly from the inputs, which is frozen, indexing via a float won't run into any precision issues?

Reply:
Yeah, that is my thought. If we run into trouble later, we can truncate the float or something